control/Controller.php

```diff
@@ -463,13 +463,20 @@ public function popCurrent() {
 	 */
 	public function redirect($url, $code=302) {
 		if(!$this->response) $this->response = new SS_HTTPResponse();
-		
-		if($this->response->getHeader('Location') && $this->response->getHeader('Location') != $url) {
-			user_error("Already directed to " . $this->response->getHeader('Location')
-				. "; now trying to direct to $url", E_USER_WARNING);
-			return;
-		}
 
+		$location = $this->response->getHeader('Location');
+		if($location){
+			
```

Comment:
What if redirecting to protocol-less url? E.g. `$this->redirect('//google.com');`